src/mlpack/core/optimizers/adam/adam.hpp

```diff
@@ -186,6 +187,8 @@ using Nadam = AdamType<NadamUpdate>;
 
 using NadaMax = AdamType<NadaMaxUpdate>;
 
+using Adamirror = AdamType<AdamirrorUpdate>;
```

Comment:
What do you think should we call this: ``OptimAdam``, this is what they use in the paper? Same for the update rule.

Reply:
I thought OptimAdam might be confusing since many libraries refer to optimizer modules as optim. So OptimAdam, might be confused as the Adam optimizer itself. Otherwise OptimAdam would be fine. What do you think?

Reply:
Another idea is to go with ``OptimisticAdam``, to avoid the confusion.

Reply:
That sounds good. I'll change the name.